File: src/main/java/com/DavidM1A2/afraidofthedark/common/spell/component/deliveryMethod/base/ISpellDeliveryTransitioner.java
package com.DavidM1A2.afraidofthedark.common.spell.component.deliveryMethod.base;

/**
 * Interface representing a delivery method transitioner to go from one delivery method to another
 */
public interface ISpellDeliveryTransitioner
{
    /**
     * Called to deliver the effects to the target by whatever means necessary
     *
     * @param state The state of the spell to deliver
     */
    void transition(DeliveryTransitionState state);
}

File: src/main/java/com/davidm1a2/afraidofthedark/common/spell/component/deliveryMethod/base/ISpellDeliveryTransitioner.kt
package com.davidm1a2.afraidofthedark.common.spell.component.deliveryMethod.base

import com.davidm1a2.afraidofthedark.common.spell.component.DeliveryTransitionState

/**
 * Interface representing a delivery method transitioner to go from one delivery method to another
 */
interface ISpellDeliveryTransitioner {
    /**
     * Called to deliver the effects to the target by whatever means necessary
     *
     * @param state The state of the spell to deliver
     */
    fun transition(state: DeliveryTransitionState)
}